File: src/main/java/ttd/site/generator/TemplateService.java
package ttd.site.generator;

import java.util.Collection;
import java.util.Date;
import java.util.List;
import java.util.Map;

interface TemplateService {

	String monthly(YearMonth yearMonth, Map<String, List<Link>> links);

	String index(Collection<Link> links);

	String daily(Date date, Collection<Link> links);

}

File: src/main/kotlin/ttd/site/generator/TemplateService.kt
package ttd.site.generator

import java.util.*

interface TemplateService {
	fun monthly(yearMonth: YearMonth, links: Map<String, List<Link>>): String
	fun index(yearAndMonths: List<YearMonth>): String
	fun daily(date: Date, links: Collection<Link>): String
}